subprojects/docs/src/docs/userguide/troubleshootingDependencyResolution.adoc

```diff
@@ -15,83 +15,93 @@
+Conflicting versions of the same module should be detected and either resolved or cause an exception. If you don't use transitive dependency management, version conflicts are undetected and may influence the order of the classpath determining what version of a dependency will be selected. In short, version conflicts can cause classpath issues and are extremely hard to debug.
 
-If you haven't had to deal with the curse of conflicting versions of jars on a classpath, here is a small anecdote of the fun that awaits you. In a large project with 30 submodules, adding a dependency to a subproject changed the order of a classpath, swapping Spring 2.5 for an older 2.4 version. While the build continued to work, developers were starting to notice all sorts of surprising (and surprisingly awful) bugs in production. Worse yet, this unintentional downgrade of Spring introduced several security vulnerabilities into the system, which now required a full security audit throughout the organization.
-
-In short, version conflicts are bad, and you should manage your transitive dependencies to avoid them. You might also want to learn where conflicting versions are used and consolidate on a particular version of a dependency across your organization. With a good conflict reporting tool like Gradle, that information can be used to communicate with the entire organization and standardize on a single version. _If you think version conflicts don't happen to you, think again._ It is very common for different first-level dependencies to rely on a range of different overlapping versions for other dependencies, and the JVM doesn't yet offer an easy way to have different versions of the same jar in the classpath (see <<sub:dependency_management_and_java>>).
-
-Gradle offers the following conflict resolution strategies:
+Gradle can <<managing_transitive_dependencies,manage transitive dependencies>> for you with the help of the following conflict resolution strategies:
 
 * _Newest_: The newest version of the dependency is used. This is Gradle's default strategy, and is often an appropriate choice as long as versions are backwards-compatible.
-* _Fail_: A version conflict results in a build failure. This strategy requires all version conflicts to be resolved explicitly in the build script. See api:org.gradle.api.artifacts.ResolutionStrategy[] for details on how to explicitly choose a particular version.
+* _Fail_: A version conflict results in a build failure. This strategy requires all version conflicts to be resolved explicitly in the build script. Fore more information, see api:org.gradle.api.artifacts.ResolutionStrategy#failOnVersionConflict()[].
```

Comment:
We _really_ ought to show and link to a real build scan here with an image showing what was requested and what was resolved.

Reply:
I am mentioning build scans in a sentence a little bit below in the same paragraph.

> link:https://scans.gradle.com/get-started[Build scans] and <<inspecting_dependencies,dependency reports>> are immensely helpful in identifying and fixing dependency version conflicts. Make sure you are familiar with those tools!

I don't think build scans will help much here. It just fails the build probably saying _that_ there was a version conflict. You'd see the same behavior as in the console. Are you thinking about something specific here?

Reply:
I would put that also in a rewritten first paragraph (see comments above). So maybe just redo all the above as one or two paragraphs without the list:

- Gradle resolves conflict by picking the highest version
- If you want more details where this version came from, you should use build scans (and dependencyInsights if we want to mention)
- If you want to be more strict about everything, you can tell Gradle to fail

Reply:
Makes sense. done